### PR TITLE
Feature group chat

### PR DESCRIPTION
**Pull request by brendenmcfarlane**
Created Interface adapter layer for usecases Retrieve Friends List and Create Group Chat. I have also created tests for the output adapters, as well as refactored the code at the usecase layer to correctly retrieve the IP address and peer port. I will create controller tests and improve documentation at a later time.

### CODE REVIEW

**Review by davidlyj — APPROVED** [review]
Looks good and ready to be merged.

File: src/main/java/server/interface_adapters/create_gc/CreateGCOutputAdapter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     *
+     * String contains Constants.GROUP_CHAT, the admins UID, every members UID,
+     * in that order separated by Constants.SPR
+     */
+    @Override
+    public void prepareSuccessView(CreateGCOutputData output_data) {
+        String peer_id = output_data.getPeerID();
+        int peer_port = output_data.getPeerPort();
+        String msg = String.valueOf(Constants.GROUP_CHAT);
+        msg = msg + Constants.SPR + String.valueOf(output_data.getAdmin());
```
Comment (xingjianll) [inline src/main/java/server/interface_adapters/create_gc/CreateGCOutputAdapter.java:23]: `msg = Constants.GROUP_CHAT + Constants.SPR + String.valueOf(output_data.getAdmin()` would be a simpler implementation of this line.

File: src/main/java/server/usecases/create_gc/CreateGCInteractor.java [hunk 2]
```diff
@@ -46,8 +46,10 @@ public void create(CreateGCInputData input_data) {
         }
         GroupChat gc = new GroupChat(this.database.getNewUID(), admin, members);
         database.addGC(gc);
+        int peerPort = database.getPeerPortFromUID(input_data.getAdmin());
+        String peerID = database.getPeerIDFromUID(input_data.getAdmin());
         CreateGCOutputData output = new CreateGCOutputData(
-                input_data.getAdmin(), input_data.getMembers(), input_data.getTime());
+                input_data.getAdmin(), input_data.getMembers(), peerID, peerPort);
```
Comment (xingjianll) [inline src/main/java/server/usecases/create_gc/CreateGCInteractor.java:52]: What about the name of the groupchat? A chat has uid, name description, I think at least the uid needs to be sent to the clients

Comment (xingjianll) [reply src/main/java/server/usecases/create_gc/CreateGCInteractor.java:52]: Also, you also need to send the groupchat info to other members of the group chat, not just the admin, for example, a user can input `$createchat <chat name> <member uid1> <member uid2> ....` and when a group chat is created the information of the chat will be sent to all members including the admin (I assume the admin is also the person who created the chat?)

**Review by xingjianll — CHANGES_REQUESTED** [review]
Good work! Please check the code review comments. Also, use the database's `newChat()` method to make a new chat in the database. I believe the 4th cell (or the 5th i cant remember) in chat.csv is used to store all the member ids (in the format `1-5-2-3-15-26254-123 ...`), make sure the first integer is the admin.

**Pull request merged**